File: common/src/test/scala/com/taskadapter/connector/common/FieldPrettyNameBuilderTest.scala
package com.taskadapter.connector.common

import com.taskadapter.model.{CustomDate, Priority, Summary}
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

@RunWith(classOf[JUnitRunner])
class FieldPrettyNameBuilderTest extends FunSpec with Matchers {
  describe("pretty name generator") {
    it("custom field") {
      FieldPrettyNameBuilder.getPrettyFieldName(CustomDate("Actual Finish")) shouldBe "CustomDate(Actual Finish)"
    }
    it("regular field") {
      FieldPrettyNameBuilder.getPrettyFieldName(Summary) shouldBe "Summary"
    }
    it("priority") {
      FieldPrettyNameBuilder.getPrettyFieldName(Priority) shouldBe "Priority"
    }
  }
}

File: common/src/test/java/com/taskadapter/connector/common/FieldPrettyNameBuilderTest.java
package com.taskadapter.connector.common;

import com.taskadapter.model.AllFields;
import com.taskadapter.model.CustomDate;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class FieldPrettyNameBuilderTest {
    @Test
    public void customField() {
        assertThat(FieldPrettyNameBuilder.getPrettyFieldName(new CustomDate("Actual Finish")))
                .isEqualTo("CustomDate(Actual Finish)");
    }

    @Test
    public void regularField() {
        assertThat(FieldPrettyNameBuilder.getPrettyFieldName(AllFields.summary))
                .isEqualTo("Summary");
    }

    @Test
    public void priority() {
        assertThat(FieldPrettyNameBuilder.getPrettyFieldName(AllFields.priority))
                .isEqualTo("Priority");
    }
}
